tests/go-validation/v1apitests/provisioning_test.go

```diff
@@ -0,0 +1,92 @@
+package v1_api_tests
+
+import (
+	"fmt"
+	"testing"
+
+	apisV1 "github.com/rancher/rancher/pkg/apis/provisioning.cattle.io/v1"
+	v1 "github.com/rancher/rancher/pkg/generated/norman/core/v1"
+	"github.com/rancher/rancher/tests/go-validation/rancherrest"
+	"github.com/rancher/rancher/tests/go-validation/utils"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+const (
+	digitalOceanCloudCredentialName = "docloudcredential"
+	namespace                       = "fleet-default"
+	defaultRandStringLength         = 5
+	baseDOClusterName               = "docluster"
+)
+
+func TestProvisioning_DigitalOcean(t *testing.T) {
+	t.Log("Create Cloud Credential")
+	client, err := rancherrest.NewClient()
+	require.NoError(t, err)
+
+	cloudCredentialName := digitalOceanCloudCredentialName + utils.RandStringBytes(defaultRandStringLength)
+	doCloudCred := rancherrest.NewDigitalOceanCloudCredential(cloudCredentialName, "", namespace)
+
+	_, err = client.CreateCloudCredential(doCloudCred)
+	require.NoError(t, err)
+	t.Log("Cloud Credential was created successfully")
+
+	//randominize name
+	clusterName := baseDOClusterName + utils.RandStringBytes(defaultRandStringLength)
+
+	generatedPoolName := fmt.Sprintf("nc-%s-pool1", clusterName)
+
+	machinePoolConfig := rancherrest.NewMachinePoolConfig(generatedPoolName, rancherrest.DOKind, namespace, rancherrest.DOPoolType, "ubuntu-20-04-x64", "sfo3", "s-2vcpu-4gb")
+
+	t.Logf("Creating DO machine pool config %s", generatedPoolName)
+	podConfigClient := client.NewPodConfigClient(rancherrest.DOResourceConfig)
+
+	machinConfigResult, err := podConfigClient.CreateMachineConfigPool(machinePoolConfig)
+	require.NoError(t, err)
+
+	t.Logf("Successfully created DO machine pool %s", generatedPoolName)
+
+	provistioningClient, err := rancherrest.NewProvisioningClient()
+	require.NoError(t, err)
+
+	machinePool := rancherrest.MachinePoolSetup(true, true, true, "pool1", 1, machinConfigResult)
+	machinePools := []apisV1.RKEMachinePool{
+		machinePool,
+	}
+	clusterConfig := rancherrest.NewClusterConfig(clusterName, namespace, "cilium", machinConfigResult.GetName(), cloudCredentialName, machinePools)
+
+	cluster := provistioningClient.NewCluster(namespace)
+
+	t.Logf("Creating Cluster %s", clusterName)
+	v1Cluster, err := cluster.CreateCluster(clusterConfig)
+	require.NoError(t, err)
+
+	assert.Equal(t, v1Cluster.Name, clusterName)
+
+	t.Logf("Created Cluster %s", v1Cluster.ClusterName)
+
+	t.Logf("Checking status of cluster %s", v1Cluster.ClusterName)
+	//check cluster status
+	ready, err := cluster.CheckClusterStatus(clusterName)
+	assert.NoError(t, err)
+	assert.True(t, ready)
+
+	if utils.RancherCleanup() {
+		err = ClearDigitalOceanCluster(client, doCloudCred, cluster, clusterName)
+		require.NoError(t, err)
```

Comment:
Can there be a case where the `ClearCluster` removes the cluster from Rancher but still the resources are not removed? Can we wait somehow to check the resources are actually cleared? 

I'm curious because removing manually from Dashboard the reference gets almost immediately removed but the resource removal takes more time to clear the resources.

Reply:
We can do that I can have different clean up methods for cleaning up resources vs clusters. I did it this way, because I create a cloud credential before every test. I'll do whatever makes the most sense. I can have a check to see if a cloud cred exists beforehand, if that works better.

Reply:
In terms of actually checking the resources like DO, maybe that's a future thing ... but I'll do whatever the team wants. I'm sure I would need to setup a DO api client for example

Reply:
My 2 cents: I agree that we would want to verify that the resources are actually removed, but I think it's fine as-is for a first pass.